agent/command/git.go

```diff
@@ -870,15 +871,29 @@ func (c *gitFetchProject) fetch(ctx context.Context,
 		}
 	}
 
-	// Clone the project's modules.
+	// We use context.Background() instead of ctx here because
+	// using ctx ruins the honeycomb traces (the require the same
+	// uncancelled context).
+	g, gCtx := errgroup.WithContext(context.Background())
+	g.SetLimit(10)
+
+	// Clone the project's modules in goroutines.
 	for _, moduleName := range conf.BuildVariant.Modules {
-		if err := ctx.Err(); err != nil {
-			return errors.Wrapf(err, "canceled while applying module '%s'", moduleName)
-		}
-		err = c.fetchModuleSource(ctx, comm, conf, logger, jpm, td, opts.token, opts.method, p, moduleName)
-		if err != nil {
-			return errors.Wrapf(err, "fetching module source '%s'", moduleName)
-		}
+		moduleName := moduleName
+		g.Go(func() error {
+			if err := ctx.Err(); err != nil {
+				return errors.Wrapf(err, "canceled while applying module '%s'", moduleName)
+			}
+			// If gCtx has an error, this goroutine should not run and the error will be processed in the Wait() call.
+			if err := gCtx.Err(); err != nil {
```

Comment:
Doesn't this duplicate the above stanza?

Reply:
The above stanza is related to our context, which might be cancelled by any reason. While this stanza is cancelled when any other goroutine (g.Go call) returns an error.

Reply:
Ah, sorry, I missed the `ctx`/`gCtx` distinction. I'm still skeptical that we need a new context, but we can discuss in the other thread.